bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/wrapper/VehiclePositionWrapper.java

```diff
@@ -46,21 +56,15 @@ private State getPositionAsState(PositionData details) {
     }
 
     public State getPosition() {
-        if (vehicle.position.latitude != null) {
-            return getPositionAsState(vehicle.position);
-        } else if (vehicle.calculatedPosition.latitude != null) {
-            return getPositionAsState(vehicle.calculatedPosition);
-        }
-        return UnDefType.NULL;
+        return position.map(pos -> getPositionAsState(pos)).orElse(UnDefType.UNDEF);
     }
 
     public @Nullable String getPositionAsJSon() {
-        PositionData details = vehicle.position;
-        if (details != null && details.latitude != null && details.longitude != null) {
+        if (getPosition() != UnDefType.UNDEF) {
```

Comment:
Isn't this equivalent?
 ```suggestion
        if (position.isPresent()) {
```

Reply:
No because it also checks the presence of underpinned elements (latitude & longitude)